lib/sources/s3/agent.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();
+
+Object.defineProperty(exports, "__esModule", {
+  value: true
+});
+
+var _awsSdk = require('aws-sdk');
+
+var AWS = _interopRequireWildcard(_awsSdk);
+
+function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }
+
+function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
+
+var Agent = function () {
+  /**
+   * @param bucket {string}
+   * @param path {string}
+   */
+
+  function Agent(bucket, path) {
+    _classCallCheck(this, Agent);
+
+    this._bucket = bucket;
+    this._path = path;
+    this._s3 = new AWS.S3();
+  }
+
+  /**
+   * Generates query params for S3
+   * @param eTag
```

Comment:
Is the `eTag` parameter a string, or does the SDK map it to some kind of structure?